src/Adapter/TaxRulesGroup/CommandHandler/EditTaxRulesGroupHandler.php

```diff
@@ -0,0 +1,83 @@
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+namespace PrestaShop\PrestaShop\Adapter\TaxRulesGroup\CommandHandler;
+
+use PrestaShop\PrestaShop\Adapter\TaxRulesGroup\AbstractTaxRulesGroupHandler;
+use PrestaShop\PrestaShop\Core\Domain\TaxRulesGroup\Command\EditTaxRulesGroupCommand;
+use PrestaShop\PrestaShop\Core\Domain\TaxRulesGroup\CommandHandler\EditTaxRulesGroupHandlerInterface;
+use PrestaShop\PrestaShop\Core\Domain\TaxRulesGroup\Exception\CannotUpdateTaxRulesGroupException;
+use PrestaShop\PrestaShop\Core\Domain\TaxRulesGroup\Exception\TaxRulesGroupException;
+use PrestaShopException;
+
+/**
+ * Handles tax rules group edition
+ */
+class EditTaxRulesGroupHandler extends AbstractTaxRulesGroupHandler implements EditTaxRulesGroupHandlerInterface
+{
+    /**
+     * {@inheritdoc}
+     *
+     * @throws CannotUpdateTaxRulesGroupException
+     * @throws TaxRulesGroupException
+     */
+    public function handle(EditTaxRulesGroupCommand $command): void
+    {
+        $taxRulesGroup = $this->getTaxRulesGroup($command->getTaxRulesGroupId());
+
+        if (null !== $command->getName()) {
+            $taxRulesGroup->name = $command->getName();
+        }
+        if (null !== $command->isEnabled()) {
+            $taxRulesGroup->active = $command->isEnabled();
+        }
+
+        try {
+            if (false === $taxRulesGroup->validateFields(false)
+                || false === $taxRulesGroup->validateFieldsLang(false)) {
+                throw new TaxRulesGroupException('Tax Rules Group contains invalid field values');
+            }
+            if (false === $taxRulesGroup->update()) {
+                throw new CannotUpdateTaxRulesGroupException(
+                    sprintf(
+                        'Failed to update cms page with id %s',
+                        $command->getTaxRulesGroupId()->getValue()
+                    )
+                );
+            }
+            if (null !== $command->getShopAssociation()) {
+                $this->associateWithShops($taxRulesGroup, $command->getShopAssociation());
+            }
+
+            /* @phpstan-ignore-next-line */
+            if (!$taxRulesGroup->update()) {
+                throw new TaxRulesGroupException(sprintf('Cannot update tax with id "%s"', $taxRulesGroup->id));
+            }
+        } catch (PrestaShopException $e) {
+            throw new TaxRulesGroupException(sprintf('Cannot update tax with id "%s"', $taxRulesGroup->id));
+        }
```

Comment:
big part of this block could be moved out to a dedicated TaxRuleGroupRepository, which would reuse AbstractObjectModelRepository and a dedicated TaxRuleGroupValidator (there are plenty of examples) wouldn't require much time to implement.
We have some docs about them here https://devdocs.prestashop-project.org/8/development/architecture/domain/domain-services/ (though the thing about `Adapter` ns is  probably outdated since this ADR https://github.com/PrestaShop/ADR/blob/master/0014-drop-the-no-use-of-legacy-in-new-code-restriction.md)

Reply:
@zuk3975 Have you an example to give, please ?

Reply:
Check usages of AbstractObjetModelRepository|AbstractObjectModelValidator and you'll see plenty of them :wink:

Reply:
@zuk3975 It's done in 4013a4b. I would you like your advice.